src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -22,27 +23,28 @@ const helpers = require('../lib/helpers');
 const getCompany = async (request, h) => {
   const { regionId, companyId } = request.params;
   const { licenceId, redirectPath } = request.query;
-  const { company } = request.pre;
+  const { company, companies } = request.pre;
   // licenceId is an optional query param. if supplied it will be displayed as the caption on the forms for the session
   const { licenceNumber } = licenceId ? await dataService.getLicenceById(licenceId) : { licenceNumber: null };
   // The company name and licence number set here will be used in the select address page
   const data = { viewData: { redirectPath, licenceNumber, licenceId, companyName: titleCase(company.name) } };
   const session = dataService.sessionManager(request, regionId, companyId, data);
-  const selectedCompany = has(session, 'agent') && isEmpty(session.agent) ? company : helpers.getAgentCompany(session);
+  const selectedCompany = has(session, 'agent') && isEmpty(session.agent) ? company : await helpers.getAgentCompany(session);
```

Comment:
The loadCompanies pre handler already fetches the agent company if it exists in the session. I would rather leave it here than in the pre-handler because having an agent is more of an exception than not so the prehandler will in most cases do less work

Reply:
I've kept the pre-handler but I tidied this up, by getting the agent using the helper above and passing the prehandler once.